be/src/exec/pipeline/fragment_executor.cpp

```diff
@@ -75,6 +75,19 @@ Status FragmentExecutor::prepare(ExecEnv* exec_env, const TExecPlanFragmentParam
 
     RETURN_IF_ERROR(exec_env->query_pool_mem_tracker()->check_mem_limit("Start execute plan fragment."));
 
+    int32_t degree_of_parallelism = 1;
+    if (request.__isset.pipeline_dop && request.pipeline_dop > 0) {
+        degree_of_parallelism = request.pipeline_dop;
+    } else {
+        // default dop is a half of the number of hardware threads.
+        degree_of_parallelism = std::max<int32_t>(1, std::thread::hardware_concurrency() / 2);
+    }
+
+    if (exec_env->query_context_mgr()->is_overloaded(degree_of_parallelism)) {
+        return Status::Corruption(strings::Substitute("BE has overloaded with $0 tasks",
```

Comment:
suggest to use InternalError or RuntimeError,  Corruption means that data intergrity is broken.